app/src/openxr/cpp/OpenXRLayers.cpp

```diff
@@ -139,14 +139,13 @@ OpenXRLayerCube::Init(JNIEnv * aEnv, XrSession session, vrb::RenderContextPtr& a
   OpenXRLayerBase<VRLayerCubePtr, XrCompositionLayerCubeKHR>::Init(aEnv, session, aContext);
 }
 
-void
-OpenXRLayerCube::Destroy() {
+OpenXRLayerCube::~OpenXRLayerCube() {
   if (!swapchain) {
     return;
   }
   layer->SetTextureHandle(0);
   layer->SetLoaded(false);
-  OpenXRLayerBase<VRLayerCubePtr, XrCompositionLayerCubeKHR>::Destroy();
+  destroySurface();
```

Comment:
I'm sure there is a reason for why you did it this way, could you explain why we need to notify the destruction of the layer explicitly instead of waiting for the OpenXRLayerBase to do that for us on destruction?